apps/aehttp/src/aehttp_dispatch_int.erl

```diff
@@ -278,6 +299,36 @@ get_block_txs_count(Fun) when is_function(Fun, 0) ->
             Err
     end.
 
+get_block_tx_by_index(Fun, Index, Req) when is_function(Fun, 0) ->
+    case get_block_from_chain(Fun) of
+        {ok, Block} ->
+            Txs = aec_blocks:txs(Block),
+            case length(Txs) < Index orelse Index =< 0 of
```

Comment:
A `try lists:nth(Index, Txs) of ... catch` could work here. It would eliminate the O(n) `length/1` check.